state/pruningHandler.go

```diff
@@ -0,0 +1,17 @@
+package state
+
+type pruningHandler struct {
+	isPruningEnabled bool
+}
+
+// NewPruningHandler returns a new instance of pruningHandler with the given parameters
+func NewPruningHandler(isPruningEnabled bool) *pruningHandler {
```

Comment:
Why not creating 2 constructors instead of one that requires a bool passing?
Something like: 
```GO
NewEnabledPruningHandler
NewDisabledPruningHandler
```

Reply:
Was thinking that if in the future we add more options, the number of constructors will grow exponentially :smile: